### DISCUSSION TITLE
How to adjust slider window on custom Content Query pagination?

### DISCUSSION THREAD

**Discussion by sarahessex** [statamic/cms#5582] [Q&A] [answered]
Hi, I have a custom Content Query that I am using the [->paginate()](https://statamic.dev/content-queries#paginating) method on..

````
    $paginator =  Entry::query()
            // ...custom query here...
            ->paginate(15);
````

I'm then generating the pagination data (which I copied from `Statamic\Tags\Concerns\OutputsItems->getPaginationData()`)

````
    $pagination = [
                'total_items'    => $paginator->total(),
                'items_per_page' => $paginator->perPage(),
                'total_pages'    => $paginator->lastPage(),
                'current_page'   => $paginator->currentPage(),
                'prev_page'      => $paginator->previousPageUrl(),
                'next_page'      => $paginator->nextPageUrl(),
                'auto_links'     => $paginator->render('pagination::default'),
                'links'          => $paginator->renderArray()
            ];

    $view->with( 'pagination', $pagination);
````

And in my Antlers template...

`{{ partial:pagination :paginate='pagination'}}`

This gives me the paginate so that I can use it in the same way as in your [Collection Example](https://statamic.dev/tags/collection#pagination-examples)...

````
    {{ paginate }}
    <ul class="pagination">
        {{ if prev_page }}
            <li><a href="{{ prev_page }}">&laquo;</a></li>
        {{ else }}
            <li class="disabled"><span>&laquo;</span></li>
        {{ /if }}
 
        {{ links:segments }}
 
            {{ first }}
                {{ if page == current_page }}
                    <li class="active"><span>{{ page }}</span></li>
                {{ else }}
                    <li><a href="{{ url }}">{{ page }}</a></li>
                {{ /if }}
            {{ /first }}
 
            {{ if slider }}
                <li class="disabled"><span>...</span></li>
            {{ /if }}
 
            {{ slider }}
                {{ if page == current_page }}
                    <li class="active"><span>{{ page }}</span></li>
                {{ else }}
                    <li><a href="{{ url }}">{{ page }}</a></li>
                {{ /if }}
            {{ /slider }}
 
            {{ if slider || (!slider && last) }}
                <li class="disabled"><span>...</span></li>
            {{ /if }}
 
            {{ last }}
                {{ if page == current_page }}
                    <li class="active"><span>{{ page }}</span></li>
                {{ else }}
                    <li><a href="{{ url }}">{{ page }}</a></li>
                {{ /if }}
            {{ /last }}
 
        {{ /links:segments }}
 
        {{ if next_page }}
            <li><a href="{{ next_page }}">&raquo;</a></li>
        {{ else }}
            <li class="disabled"><span>&raquo;</span></li>
        {{ /if }}
    </ul>
    {{ /paginate }}
````

All working fine, I can either do `{{ auto_links }}` or include the above partial for custom pagination.

My question is that the default slider window is 10, is there any way I can adjust it down? I found this discussion https://github.com/statamic/cms/pull/5001 but I think that `on_each_side` usage is for the core collection pagination only? Can I insert it somewhere as a parameter to have the same effect or would it be a case of having to write something custom?

I've tried setting `$paginator->onEachSide = 3` before calling `$paginator->renderArray()` on the `links` part of my `$pagination` array, but it hasn't had an effect.

Thanks.

**Comment by jasonvarga** [comment, marked as answer]
Instead of doing `$paginator->onEachSide = 3`, you should use the method:

```php
$paginator->onEachSide(3);
```

**Reply by sarahessex** [reply]
```
            $paginator = $query->paginate($max);

            $paginator->onEachSide(3);

            $pagination = [
                'total_items'    => $paginator->total(),
                'items_per_page' => $paginator->perPage(),
                'total_pages'    => $paginator->lastPage(),
                'current_page'   => $paginator->currentPage(),
                'prev_page'      => $paginator->previousPageUrl(),
                'next_page'      => $paginator->nextPageUrl(),
                'auto_links'     => $paginator->render('pagination::default'),
                'links'          => $paginator->renderArray(),
            ];
```

Is still getting me 10 on each side. (When I'm using `{{ links:segments }}` ... `{{ /links:segments }}` in the partial example )

[Screenshot here](https://imgur.com/a/igqLlpv)

**Reply by jasonvarga** [reply]
Honestly the pagination window thing is a bit of a black box to me.

If you go to page 11, what does it look like?

**Reply by sarahessex** [reply]
I know the feeling..!

Ok, I think I've realised the onEachSide parameter is not quite what I thought it was... seems to be 'on each side of the current page number' rather than the size of the first/last link ranges (though, it does adjust that too, just not in a way I can work out)...

When I do `$paginator->onEachSide(1);`...

When on page 1:

**1** 2 3 4 5 6 ... 28 29

When on page 11:

1 2 ... 10 **11** 12 ... 28 29 `// 1 link on each side of current`

When I do `$paginator->onEachSide(3);`...

When on page 1:

**1** 2 3 4 5 6 7 8 9 10 ... 28 29

When on page 11:

1 2 ... 8 9 10 **11** 12 13 14 ... 28 29 `// 3 links on each side of current`

I don't really get how it's working out how many to show before the separator when on page 1, but putting it right down to `$paginator->onEachSide(1)` kind of gets me what I needed so I can work with that! Thanks.
